### ISSUE TITLE
move lib to src for consistency

### ISSUE THREAD

**Issue by robwormald** [ngrx/core#4]

**Comment by MikeRyanDev** [comment]
Are you wanting it to be consistent across all ngrx projects? If that's the case, we'll need to update router, router-store, effects, notify, store-devtools, and store-log-monitor as well.

**Comment by robwormald** [comment]
Yeah, I'd like to be consistent across projects and implement this proposal:

https://github.com/dherman/defense-of-dot-js/blob/master/proposal.md

TS source in /src
Output commonJS ES5 to the root
Output standard (ES) modules to /lib
Add `module` and `modules.root` to point to ESM entry point and modules

We'll implement the same in @angular and build our es module pipeline around it. 

Thoughts?

**Comment by MikeRyanDev** [comment]
Sounds good. To be clear, the ES modules we output to `/lib` need to be ES5 with ES6 modules, correct?
